### PR TITLE
add more benchmarks

### PR DESCRIPTION
**Pull request by TomTranter**

### CODE REVIEW

**Comment by TomTranter** [comment]
@brosaplanella are the benchmarks configured to run only when PR requests are merged into main? or do they run on branches too and on commits. The reason I ask is I added a very long simulation. @wigging I don't know if you want to use this feature at all to check the solvers and managers but might be useful.

**Comment by brosaplanella** [comment]
At the moment they don't run for pushes and pull requests, they are only triggered manually and as a scheduled task. We can make them run on pull requests and pushes (though I think they can only work if the PRs come from the actual repository, not forks) and probably we can split the benchmark tests so only a subset is run for PRs (and keep the slow ones for the scheduled task).

File: benchmarks/basic_benchmarks.py [hunk 1]
```diff
@@ -10,3 +11,134 @@ def setup(self):
...
+
+
+class SmallPack:
+    def setup(self):
+        self.netlist = lp.setup_circuit(Np=2, Ns=1, Rb=1e-4, Rc=1e-2)
+        chemistry = pybamm.parameter_sets.Chen2020
+        self.parameter_values = pybamm.ParameterValues(chemistry=chemistry)
+
+    def time_discharge_1cpu(self):
+        experiment = pybamm.Experiment(
```
Comment (brosaplanella) [inline benchmarks/basic_benchmarks.py:23]: Do you want to time the setting up of the experiment? If not, you can move it to setup so you are only timing the solving time.

**Review by brosaplanella — APPROVED** [review]
Looks good! Though if the initial SoC is a new feature, the benchmarks will crash for old versions and we will not be able to see the slowdown. Maybe we can add some additional tests without specifying the initial SoC?

File: benchmarks/basic_benchmarks.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 # See "Writing benchmarks" in the asv docs for more information.
 
 import liionpack as lp
+import pybamm
 
 
 class BasicBenchmark:
```
Comment (brosaplanella) [inline benchmarks/basic_benchmarks.py:8]: This test for some reason doesn't work. Is the syntax correct?

**Comment by brosaplanella** [comment]
Just manually ran the benchmarks from this branch (https://github.com/pybamm-team/liionpack/actions/runs/1544484568). You can see the increase in time in the later version (plot here: https://pybamm-team.github.io/liionpack-bench/#/). There is also some issue with the `LargePack` benchmarks which fail.

**Comment by brosaplanella** [comment]
Here is the documentation on how to run the benchmarks locally: https://asv.readthedocs.io/en/stable/using.html#running-benchmarks

File: benchmarks/basic_benchmarks.py (hunk 1 above)
Comment (TomTranter) [reply benchmarks/basic_benchmarks.py:23]: I don't think it adds much overhead compared to solve time but the idea was to vary the experiments for a given netlist so I put it in the methods. I guess I could define the different experiments in the setup and then pick one in the test

File: benchmarks/basic_benchmarks.py (hunk 2 above)
Comment (TomTranter) [reply benchmarks/basic_benchmarks.py:8]: It runs locally so I guess it times out. I will shorten it

Comment (TomTranter) [reply benchmarks/basic_benchmarks.py:8]: I have reduced the time and added timeout

**Pull request merged**